demo

```diff
@@ -43,35 +36,76 @@ execute() {
   ${DRY_RUN:-false} || "$@"
 }
 
+#mask oc and tkn to use the $NAMESPACE
+OC() {
+  oc -n ${NAMESPACE} "$@"
+}
+
+TKN() {
+ tkn -n ${NAMESPACE} "$@"
+}
+
+demo.validate() {
+  tkn version >/dev/null 2>&1 || err 1 "no tkn binary found"
+  oc version >/dev/null 2>&1 || err 1 "no oc binary found"
+  return 0
+}
+
 demo.prep() {
-  oc get ns -o name | grep -q namespace/pipelines-tutorial || {
-    oc new-project pipelines-tutorial
+  demo.validate
+  oc get ns "$NAMESPACE" 2>/dev/null  || {
```

Comment:
neat
```suggestion
oc new-project 2>/dev/null || true
```

Reply:
@chmouel wouldn't that ignore **all** errors? not just the error due to `ns` already existing? 
I didn't find a better way to ignore only the `already exists` error.  In this case, the script validates that the `ns` does not exist before it creates it, there by `new-project` failure should not be ignored using ` || true`